### ISSUE TITLE
NavigationView or something similar?

### ISSUE THREAD

**Issue by alessandrofrancesconi** [AvaloniaUI/Avalonia#2521]
Does Avalonia provides something equivalent/similar to NavigationView layout?
https://docs.microsoft.com/it-it/windows/uwp/design/controls-and-patterns/navigationview

**Comment by grokys** [comment]
Not in core Avalonia, no. I'm not sure if anyone has tried implementing it.

**Comment by robloo** [comment]
With SplitView done I think the WinUI NavigationView can be ported to C# and brought over. This is a fundamental control to several modern UWP/WinUI apps and will significantly lower the barrier of entry for applications coming over to Avalonia. I definitely this it's a control that should go into the main repo.

**Comment by robloo** [comment]
I should add, @kinnara already has a nice C# port completed here: https://github.com/Kinnara/ModernWpf/tree/master/ModernWpf.Controls/NavigationView

**Comment by robloo** [comment]
The latest hierarchical Navigation View from WinUI depends on the ItemsRepeater now. I believe Avalonia's ItemsRepeater does not support Tree structures (for good reasons). So this might create a hinderance from porting over the latest code.

This is more a note to anyone who might want to port this over -- myself included.

**Comment by maxkatz6** [comment]
SelectionModel tree structure is required, because even top and bottom lists is represented as two leafs of the selection tree. Right now it is possible to implement old variation of NavigationView, or wait for updated SelectionModel, or implement internal replacement.

**Comment by robloo** [comment]
@maxkatz6 Yes, I was thinking we wait for the new [SelectionModel](https://github.com/AvaloniaUI/Avalonia/pull/4533#issuecomment-679849357) rather than port the old version at this point -- sorry I misspoke talking about the ItemsRepeater and SelectionModel together.

I would love to trial porting some UWP apps over to Avalonia but without NavigationView its just too much work to get them running at first. If other things don't pan out I might be targeting Avalonia sooner rather than later though and will contribute this control myself if that time comes (along with finishing up/improving ColorPicker :) ). I firmly believe Avalonia has the best architecture and is the future UI stack we will all want to be moving to.

**Comment by kvalitetskontrollMarkaryd** [comment]
We really need NavigationView!!!

**Comment by danwalmsley** [comment]
> We really need a proper NavigationView!!!

@kvalitetskontrollMarkaryd 
If you or someone else wants to port that one from modern wpf it would be accepted .

**Comment by kvalitetskontrollMarkaryd** [comment]
I don't have the required skills. 

I have found this: https://github.com/amwx/FluentAvalonia

Looks promising!

**Comment by robloo** [comment]
I'm curious if there has been any recent discussions about NavigationView. The Fluent Avalonia port is the de-facto Avalonia version of this control now and it would be great to pull it into mainline Avalonia itself.

I know there was discussion with the core team about this in the past based on [this comment](https://github.com/AvaloniaUI/Avalonia/issues/6269#issuecomment-882847209). Is there any chance after @amwx does his round of updates for Windows 11 this can be pulled into Avalonia?

>>@amwx Is awesome! I do hope the FontIcons, NavigationView, updated themes, etc find their way into Avalonia itself. This has been discussed but not with the core maintainers afaik.
>
>We had a discussion about the NavigationView a couple months ago. Not much movement on that since though b/c I've been working on some other things and there's a few things I'd like to clean up/fix with it before (and if) it gets moved (like supporting multiple datatemplates, which is what I'm trying to work out now). @maxkatz6 is working on Icons and his approach is probably better for the core repo since it's more flexible.

**Comment by alessandrofrancesconi** [comment]
As of today, I've started using the NavigationDrawer from [Material.Avalonia](https://github.com/AvaloniaCommunity/Material.Avalonia), for both Desktop and Android platforms. It does the job I needed.
